contracts/liquidity_pool.cairo

```diff
@@ -330,11 +483,203 @@ end
     # for short put:
         # return amount - (max(0, amount * (strike_price - current_price)) in ETH)
 func expire_option_token{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    currency_address:felt,
+    option_token_address: felt,
+    option_type: felt,
+    option_side: felt,
+    strike_price: felt,
+    underlying_price: felt,
+    amount: felt,
+    maturity: felt
+):
+    # FIXME: tbd
+    alloc_locals
+
+    # Make sure the contract is the one that user wishes to expire
+    let (contract_option_type) = IOptionToken.option_type(option_token_address)
+    let (contract_strike) = IOptionToken.strike(option_token_address)
+    let (contract_maturity) = IOptionToken.maturity(option_token_address)
+    let (contract_option_side) = IOptionToken.side(option_token_address)
+    let (current_block_time) = get_block_timestamp()
+
+    with_attr error_message("Required contract doesn't match the address."):
+        assert contract_option_type = option_type
+        assert contract_strike = strike
+        assert contract_maturity = maturity
+        assert contract_option_side = side
+    end
+
+    # Assert that the contract is ready to expire
+    let (is_ripe) = is_le(maturity, current_block_time)
+    with_attr error_message("Contract isn't ripe yet."):
+        assert is_ripe
+    end
+
+
+    if option_side == TRADE_SIDE_LONG:
+        _expire_option_token_long(
+            currency_address = currency_address,
+            option_token_address = option_token_address,
+            amount = amount,
+            option_type = option_type,
+            strike_price = strike_price,
+            underlying_price = underlying_price,
+        )
+
+    else:
+        _expire_option_token_short(
+            currency_address = currency_address,
+            option_token_address = option_token_address,
+            amount = amount,
+            option_type = option_type,
+            strike_price = strike_price,
+            underlying_price = underlying_price,
+        )
+
+    end
+
+    return ()
+end
```

Comment:
this looks too repetitive and does not deal with the pool...

simplified quite a lot
```
func expire_option_token
   long_value, short_value = split_option_locked_capital(option_type, strike, amout,...)

   if option_side == LONG:
      IERC20.transferFrom(..., user_address, long_value)
      adjust_pool_available_capital(short_value)
   else:
      IERC20.transferFrom(..., user_address, short_value)
      adjust_pool_available_capital(long_value)
   end
end

func split_option_locked_capita...
       if option_type == OPTION_CALL:
        # User receives -> max(0, amount * (current_price - strike_price)) in base token

        let price_diff = underlying_price - strike_price
        let to_be_paid_quote = amount * price_diff
        let to_be_paid_base = to_be_paid / underlying_price
        let to_be_paid_buyer = max(0, to_be_paid_base)
        
        let to_be_paid_seller = option_size * strike_price - to_be_paid_buyer

        return  (to_be_paid_buyer, to_be_paid_seller)
   else
      ...
   end
end
```